provider/resources.go

```diff
@@ -1891,6 +1891,14 @@ func Provider() tfbridge.ProviderInfo {
 						Transform: strings.ToLower,
 					}),
 				},
+				TransformFromState: func(_ context.Context, pm resource.PropertyMap) (resource.PropertyMap, error) {
+					// if the defaultOutboundAccessEnabled property is not set, set it to the default value of true
+					// this prevents an unnecessary replacement when upgrading the provider
+					if _, ok := pm["defaultOutboundAccessEnabled"]; !ok {
+						pm["defaultOutboundAccessEnabled"] = resource.NewBoolProperty(true)
+					}
+					return pm, nil
+				},
```

Comment:
Wasn't this included in https://github.com/pulumi/pulumi-azure/pull/2325? A rebase onto master might show the effects of this PR more clearly.

Reply:
I've merged 2325 here, not in master - I didn't want to consider the effects of 2325 under non-PRC.